dashboard/test/lib/queries/user/expired_deleted_accounts_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helper'
+
+class Queries::User::ExpiredDeletedAccountsTest < ActiveSupport::TestCase
+  describe '#call' do
+    let(:described_instance) {described_class.new(deleted_before: deleted_before)}
+    let(:deleted_before) {28.days.ago}
+    let(:expected_user) {create(:user, :deleted)}
+    subject(:expired_deleted_accounts) {described_instance.call}
+
+    it 'does not return users who are not soft-deleted' do
+      active_user = create(:user, current_sign_in_at: Time.now)
+      _(expired_deleted_accounts).wont_include active_user
+    end
+
+    it 'returns users who have been soft-deleted before the specified date' do
+      expected_user.update(deleted_at: deleted_before - 1.day)
+      _(expired_deleted_accounts).must_include expected_user
+    end
+
+    it 'does not return users who have been soft-deleted after the specified date' do
+      _(expired_deleted_accounts).wont_include expected_user
+    end
+
+    context 'when user already scrubbed of PII' do
+      it 'does not return the user' do
+        expected_user.update(deleted_at: deleted_before - 1.day)
+        create :user_data_retention_status, user: expected_user, pii_scrubbed_at: Time.now
+        _(expired_deleted_accounts).wont_include expected_user
+      end
+    end
```

Comment:
A `context` block should usually be justified by a meaningful change from the "happy path" and should apply to all the test cases within it:
```suggestion
    context 'when user already scrubbed of PII' do
      let!(:expected_user) {create(:user, :deleted, deleted_at: deleted_before - 1.day)}
 
      before do
        create :user_data_retention_status, user: expected_user, pii_scrubbed_at: Time.now
      end
      
      it 'does not return the user' do
        _expired_deleted_accounts.wont_include expected_user
      end
    end
```